ga4gh/decorators.py

```diff
@@ -0,0 +1,92 @@
+"""
+General purpose decorators
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import functools
+import signal
+import time
+
+
+class TimeoutException(Exception):
+    """
+    A process has taken too long to execute
+    """
+
+
+class Repeat(object):
+    """
+    A decorator to use for repeating a tagged function.
```

Comment:
It can be quite tricky in my experience have two processes and waiting for sockets to become ready. But you're right, it might be more hacking that it's worth to do it in a single process.